src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -73,6 +99,31 @@ HRESULT CLoggerService::GetLoggingFlags(_Out_ LoggingFlags* pLoggingFlags)
     return S_OK;
 }
 
+HRESULT CLoggerService::UpdateInstrumentationMethodLoggingFlags(_In_ GUID classId, _In_ LoggingFlags loggingFlags)
+{
+    HRESULT hr = S_OK;
+
+    IfNotInitRetUnexpected(m_initialize);
+
+    CCriticalSectionHolder holder(&m_cs);
+
+    m_instrumentationMethodFlags = LoggingFlags_None;
+
+    IfFailRetNoLog(UpdateFlags(classId, loggingFlags));
+
+    for (unordered_map<LoggingFlags, set<GUID>>::iterator it = m_loggingFlagsToInstrumentationMethodsMap.begin();
+        it != m_loggingFlagsToInstrumentationMethodsMap.end();
+        it++)
+    {
+        if (!(it->second.empty()))
```

Comment:
>if (!(it->second.empty())) [](start = 8, length = 26)

I'll add a comment here, effectively we're iterating through all of the logging flags and if the set of guids is not empty (ie. at least one instrumentation method requires that logging level), we will update the m_instrumentationMethodFlags.